Products/ZenUI3/browser/eventconsole/export.py

```diff
@@ -117,19 +127,21 @@ def csv(self, response, archive, **params):
                 if field in ("lastTime", "firstTime") and val:
                     val = datetime.utcfromtimestamp(val).isoformat()
                 elif field == DETAILS_KEY and val:
-                    # add all details in one column, otherwise for events with lots of details
-                    # the number of columns in the csv can get too large (ZEN-23871)
+                    # ZEN-ZEN-23871: add all details in one column
                     val = json.dumps(val)
                 elif not (val or val is 0) and details:
-                    # ZEN-27617: fill in value for requested field not in evt but in details
-                    val = details.get(field,'')
-                data.append(str(val).replace('\n',' ').strip() if val or val is 0 else '')
+                    # ZEN-27617: fill in value for requested field in details
+                    val = details.get(field, '')
+                if (val or val is 0):
+                    data.append(str(val).replace('\n', ' ').strip())
+                else:
+                    data.append('')
```

Comment:
perhaps this refactor instead of an `if` statement:
```
    data.append(
        str(val).replace('\n', ' ').strip()
        if (val or val == 0) else ''
    )
```